contracts/PositionToken.sol

```diff
@@ -3,20 +3,64 @@ pragma solidity 0.8.13;
 
 import "hardhat/console.sol";
 
-import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
-import {ERC721Burnable} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
 
-contract PositionToken is ERC721, ERC721Burnable {
-    constructor() ERC721("PositionToken", "PNT") {}
+contract PositionToken is ERC721, ERC721Burnable, AccessControl {
+    using Counters for Counters.Counter;
```

Comment:
Core ideas here generated using https://wizard.openzeppelin.com/#erc721 after finding the recommendation to do so in OpenZepplin docs.